pkg/controller/secret.go

```diff
@@ -0,0 +1,67 @@
+package controller
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	operatorv1alpha1 "github.com/openshift/external-secrets-operator/api/v1alpha1"
+	"github.com/openshift/external-secrets-operator/pkg/operator/assets"
+)
+
+func (r *ExternalSecretsReconciler) createOrApplySecret(es *operatorv1alpha1.ExternalSecrets, resourceLabels map[string]string, recon bool) error {
+	// secrets are only created if isCertManagerConfigEnabled disabled
+	if isCertManagerConfigEnabled(es) {
+		return nil
+	}
+
+	desired, err := r.getSecretObject(es, resourceLabels)
+	if err != nil {
+		return fmt.Errorf("failed to generate secret resource for creation: %w", err)
+	}
+
+	secretName := fmt.Sprintf("%s/%s", desired.GetNamespace(), desired.GetName())
+	r.log.V(4).Info("reconciling secret resource", "name", secretName)
+	fetched := &corev1.Secret{}
+	key := types.NamespacedName{
+		Name:      desired.GetName(),
+		Namespace: desired.GetNamespace(),
+	}
+
+	exist, err := r.Exists(r.ctx, key, fetched)
+	if err != nil {
+		return FromClientError(err, "failed to check %s secret resource already exists", secretName)
+	}
+
+	if exist && recon {
+		r.eventRecorder.Eventf(es, corev1.EventTypeWarning, "ResourceAlreadyExists", "%s secret resource already exists, maybe from previous installation", secretName)
+	}
+
+	if exist && hasObjectChanged(desired, fetched) {
```

Comment:
```suggestion
	if exist && objectMetadataModified(desired, fetched) {
```
Since we are only managing labels on this resource, just checking for the same would be sufficient, which otherwise will create deadlock loop. external-secrets controller trying to add certificate and operator trying to remove it.

Reply:
Okay got the point.